pkg/util/modules/modules.go

```diff
@@ -9,6 +9,16 @@ import (
 	"github.com/cortexproject/cortex/pkg/util/services"
 )
 
+type moduleVisibility int
+
+const (
+	// A public module is intended to be passed into `InitModuleServices`.
+	public moduleVisibility = iota
+
+	// A private module is not intended to be passed into `InitModuleServices`.
+	private
+)
+
```

Comment:
I don't see a need for this. Let's please use simple `bool`, like before. My suggestion was to modify terminology to move away from public/private to internal, since that's what these modules are.

Reply:
oh, you commented faster than I replied :) 

In this case, what would you suggest for the name for `PublicModuleNames`?

Reply:
Hmm, your other comment explains the thinking, and I can see why you went this route. Still, I feel like this goes too far for what is quite simple concept. Would `ModuleNames(skipInternal bool)` work for you?

Reply:
By the way, what if instead of `type moduleVisibility int` I use `type moduleVisibility bool`?

Reply:
We can use term "visible", or "userVisible", but let's just use that as a bool field in module struct, without introducing new type.

Reply:
`UserVisibleModuleNames()` is also pretty clear I think.

Reply:
> Hmm, your other comment explains the thinking, and I can see why you went this route. Still, I feel like this goes too far for what is quite simple concept. Would `ModuleNames(skipInternal bool)` work for you?

Don't you think it adds some mental power to the reader and the coder? When I read `ModuleNames(true)` it's very unclear to me what I get in return, I will probably have to have an alias variable so I can write `ModuleName(skipInternal)`, which makes the intend clearer, but I will have to define alias variable. For simple things like getting list of public modules, I feel a method with no arg is friendlier to the user.

Reply:
> We can use term "visible", or "userVisible", but let's just use that as a bool field in module struct, without introducing new type.

I think this will work, let me give it a try.

Reply:
Ok, done. Have a look and let me know. 

Please do let me know if you think it's still off; I happen think variable/method/class/whatever names are very important and I am more than welling to spend time to get the terms right to increase readability, so please don't run out of patience with me :)